pkg/workloads/console/integration/integration_test.go

```diff
@@ -264,14 +271,12 @@ var _ = Describe("Console", func() {
 
 			waitForSuccessfulReconcile(1)
 
-			job := &batchv1.Job{}
+			job = &batchv1.Job{}
 			identifier, _ = client.ObjectKeyFromObject(csl)
 			err = mgr.GetClient().Get(context.TODO(), identifier, job)
 
-			statusError, _ := err.(*errors.StatusError)
 			Expect(err).To(HaveOccurred(), "the job should remain deleted, but it has been recreated")
-			Expect(err).To(BeAssignableToTypeOf(statusError), "error should be an API error")
-			Expect(statusError.Status()).To(MatchFields(IgnoreExtras, Fields{"Code": BeEquivalentTo(404)}))
+			Expect(errors.IsNotFound(err)).To(Equal(true), "a NotFound error was expected, but got a different error")
```

Comment:
Minor note: The downside of this approach is that if the assertion fails then the error message is not going to be of much use (i.e. `true != false`); we'd have to insert some extra logging to find out what went wrong

Reply:
That's true, but I'd argue that the tests become easier to read as a result. I guess it's a balance of whether you're optimising for reading or debugging.